src/vm/dllimportcallback.cpp

```diff
@@ -1408,16 +1420,14 @@ VOID UMThunkMarshInfo::RunTimeInit()
         new (&sigInfo) PInvokeStaticSigInfo(pMD);
     else
         new (&sigInfo) PInvokeStaticSigInfo(GetSignature(), GetModule());
-
     if (sigInfo.GetCallConv() == pmCallConvCdecl)
     {
-        // caller pop
-        m_cbRetPop = 0;
+        m_cbRetPop = cbRetPop;
```

Comment:
I also except that, but it doesn't

Here is the corresponding native code:
```
Dump of assembler code for function nativeCall_RPInvoke_Vector3Ret(CallBack_RPInvoke_Vector3Ret):
   0xf7fd2180 <+0>:     push   %ebp
   0xf7fd2181 <+1>:     mov    %esp,%ebp
   0xf7fd2183 <+3>:     push   %ebx
   0xf7fd2184 <+4>:     push   %esi
   0xf7fd2185 <+5>:     sub    $0x50,%esp
   0xf7fd2188 <+8>:     call   0xf7fd218d <nativeCall_RPInvoke_Vector3Ret(CallBack_RPInvoke_Vector3Ret)+13>
   0xf7fd218d <+13>:    pop    %eax
   0xf7fd218e <+14>:    add    $0x1e73,%eax
   0xf7fd2194 <+20>:    mov    0x8(%ebp),%ecx
   0xf7fd2197 <+23>:    lea    -0x18(%ebp),%edx
   0xf7fd219a <+26>:    mov    %gs:0x14,%esi
   0xf7fd21a1 <+33>:    mov    %esi,-0xc(%ebp)
   0xf7fd21a4 <+36>:    mov    %ecx,-0x20(%ebp)
   0xf7fd21a7 <+39>:    mov    %edx,(%esp)
   0xf7fd21aa <+42>:    mov    %eax,%ebx
   0xf7fd21ac <+44>:    mov    %eax,-0x24(%ebp)
   0xf7fd21af <+47>:    call   *-0x20(%ebp)
   0xf7fd21b2 <+50>:    sub    $0x4,%esp
```

SP adjustment (``sub    $0x4,%esp``) after ``call   *-0x20(%ebp)`` (which invokes R/PInvkoe callback) implies that return buffer will be popped by callee.